cmd/extensions/oidc-policy/internal/controller/oidcpolicy_reconciler.go

```diff
@@ -26,14 +26,13 @@ import (
 
 	kuadrantv1 "github.com/kuadrant/kuadrant-operator/api/v1"
 	kuadrantv1alpha1 "github.com/kuadrant/kuadrant-operator/api/v1alpha1"
-	"github.com/kuadrant/kuadrant-operator/internal/reconcilers"
 	extcontroller "github.com/kuadrant/kuadrant-operator/pkg/extension/controller"
 	"github.com/kuadrant/kuadrant-operator/pkg/extension/types"
 	"github.com/kuadrant/kuadrant-operator/pkg/extension/utils"
 )
 
 type OIDCPolicyReconciler struct {
-	*reconcilers.BaseReconciler
+	kCtx   types.KuadrantCtx
```

Comment:
I found it convenient to attach it to the reconciler instead of passing it around.

Reply:
I like this, I wonder if there would be a way to enhance this as a default behaviour as a follow up